src/main/java/org/twilmes/sql/gremlin/processor/TraversalVisitor.java

```diff
@@ -82,15 +89,39 @@ public void visit(RelNode node) {
         final String rightAliasedColumn = rightJoinFields.get(rightKeyId);
 
         if(!(left instanceof GremlinToEnumerableConverter)) {
-            left = fieldMap.get(left).get(leftAliasedColumn);
+            if (fieldMap.get(left) != null && fieldMap.get(left).containsKey(leftAliasedColumn)) {
```

Comment:
In my own Gremlin implementation, I found left is always null when doing this simple get(leftAliasedColumn), so I have to look deeper to its joins to see what was involved in join and register them in tableTraversalMap, otherwise, later when doing get tableId, null will return and throw bad reference exception.